tests/test_maintel_apply_dof.py

```diff
@@ -0,0 +1,101 @@
+# This file is part of ts_standardscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+import random
+import unittest
+
+import numpy as np
+from lsst.ts import standardscripts
+from lsst.ts.observatory.control.maintel.mtcs import MTCS, MTCSUsages
+from lsst.ts.observatory.control.utils.enums import DOFName
+from lsst.ts.standardscripts.maintel import ApplyDOF
+
+random.seed(47)  # for set_random_lsst_dds_partition_prefix
+
+logging.basicConfig()
+
+
+class TestApplyDOF(
+    standardscripts.BaseScriptTestCase, unittest.IsolatedAsyncioTestCase
+):
+    async def basic_make_script(self, index):
+        self.script = ApplyDOF(index=index)
+
+        # Mock the MTCS
+        self.script.mtcs = MTCS(
+            domain=self.script.domain,
+            intended_usage=MTCSUsages.DryTest,
+            log=self.script.log,
+        )
+        self.mtcs.mtaos.cmd_offsetDOF.DataType = unittest.mock.PropertyMock(
```

Comment:
This should probably be `self.script.mtcs`. However, I am not sure this is needed or if this is actually what you want. Because in the script you call:

```
offset_dof_data = self.mtcs.mtaos.cmd_offsetDOF.DataType()
```

which is a method not a property. 

I think what you want to do is something like:

```
self.script.mtcs.rem.mtaos = unittest.mock.AsyncMock()
self.script.mtcs.rem.mtaos.attach_mock(
    unittest.mock.Mock(
            return_value=types.SimpleNamespace(value=np.zeros(50))
    ),
    "DataType",
)
self.script.mtcs.assert_all_enabled = unittest.mock.AsyncMock()
```

I don't think you need anything else in terms of mocking.